### PR TITLE
refactor: render `NewsReading` component directly, rather than in draft-renderer

### PR DESCRIPTION
**Pull request by nickhsine**
### Notable Changes
- add `NewsReading` and `Dropdown` components
- render `NewsReading` if `post.newsReadingGroup` is not null

### 實作細節
原本「讀報」（`NewsReading`）是放在 `@kids-reporter/draft-renderer` 透過 `draft-js` 去 render，但因為編輯回報「讀報」是在「標題之下、前言之上」，所以我把「讀報」元件從 `draft-renderer` 搬到 `frontend` 底下實作。

### CODE REVIEW

File: packages/frontend/src/app/article/[slug]/news-reading.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  className?: string
+  data: {
+    items: {
+      name: string
+      embedCode: string
+    }[]
+  }
+}
+
+const NewsReading = function ({ className, data }: NewsReadingProps) {
```
Comment (duidae) [inline packages/frontend/src/app/article/[slug]/news-reading.tsx:36]: same as above, could we use arrow function for unification

File: packages/frontend/src/app/article/[slug]/news-reading.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      items.map((r) => {
+        return {
+          name: r.name,
+          value: r.name,
+        }
+      }),
+    [items]
+  )
+  const [selectedOption, setSelectedOption] = useState(options[0])
+  const selectedReading = items.find((r) => r.name === selectedOption.value)
```
Comment (duidae) [inline packages/frontend/src/app/article/[slug]/news-reading.tsx:51]: same as above, could we add error handling for `options?.[0]` `items?.map` `items?.find`

File: packages/frontend/src/app/article/[slug]/news-reading.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+  useEffect(() => {
+    setIsMounted(true)
+  }, [])
+
+  if (!isMounted || items.length === 0) {
```
Comment (duidae) [inline packages/frontend/src/app/article/[slug]/news-reading.tsx:57]: same as above for `items?.length`

File: packages/frontend/src/app/article/[slug]/page.tsx [hunk 4]
```diff
@@ -18,6 +18,7 @@ import {
   CMS_URL,
   GetThemeFromCategory,
 } from '@/app/constants'
+import { NewsReading } from './news-reading'
```
Comment (duidae) [inline packages/frontend/src/app/article/[slug]/page.tsx:21]: `page.tsx` was modified in some places, so we may need to update the codebase & fix merge conflicts around here

**Review by duidae — COMMENTED** [review]
thanks for adding the new component! and just some minor comments as above, please have a look 🙏

File: packages/frontend/src/app/article/[slug]/dropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+type DropdownProps = {
+  className?: string
+  options: Option[]
+  onChange: (option: Option) => void
+  labelForMore?: string
+}
+
+const Dropdown: React.FC<DropdownProps> = function ({
```
[the remark this answers is not in the document]
Comment (nickhsine) [reply packages/frontend/src/app/article/[slug]/dropdown.tsx:99]: sure, we could use arrow function! I will update it in next commits.

**Comment by nickhsine** [comment]
根據建議修正了一版。

**Comment by nickhsine** [comment]
另外，我把 isMounted 移除後，就變成 server side 也會 render 了。
但因為我們還沒有 collect styled-components 在 server side render 的結果，所以畫面會有跳動。（請見影片）
這個我會再發另外一個 PR 來處理這個問題。

https://github.com/kids-reporter/kids-reporter-monorepo/assets/3000343/8070fcf3-f5b3-4f9f-bce7-568df3b9df41

**Comment by duidae** [comment]
thanks for the new commits! 
應該只剩conflict的問題(看到force push總有點害怕 😅 )
我通常都是checkout main + `git pull` -> checkout branch + `git merge master`
不知道Nick你通常都是怎麼update codebase?

**Comment by nickhsine** [comment]
conflicts 確定 PR 沒有問題，我就解一解順便 merge。

**Pull request merged**